thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/merge/AnomalyTimeBasedSummarizer.java

```diff
@@ -132,5 +136,13 @@ private static void populateMergedResult(MergedAnomalyResultDTO mergedAnomaly,
     mergedAnomaly.setStartTime(currentResult.getStartTime());
     mergedAnomaly.setEndTime(currentResult.getEndTime());
     mergedAnomaly.setCreatedTime(System.currentTimeMillis());
+    // populate current result's property as well, will be used to identify if two anomalies can be merged by comparing their property
+    mergedAnomaly.setProperties(AnomalyDetectionUtils.decodeCompactedPropertyStringToMap(currentResult.getProperties()));
+  }
+
+  // compare if two anomalies have same property when doing anomaly detection, if from same detection configuration then is mergeable
+  private static boolean isMergeable(MergedAnomalyResultDTO anomaly1, MergedAnomalyResultDTO anomaly2){
+    return ((anomaly1.getProperties() == null && anomaly2.getProperties() == null)
+        || (anomaly1.getProperties() != null && anomaly1.getProperties().equals(anomaly2.getProperties()) ));
```

Comment:
Only part of the properties need to match, not all of them. Otherwise the properties as a field will be restricted. My original design is  to use the test_config (key, value) as a unique identify of the detection config. You may following this idea and come up with a better one.